megabus/src/main/test/com/bazaarvoice/megabus/resolver/MegabusRefResolverTest.java

```diff
@@ -0,0 +1,194 @@
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.apache.kafka.streams.StreamsConfig;
+import org.apache.kafka.streams.TopologyTestDriver;
+import org.apache.kafka.streams.test.ConsumerRecordFactory;
+import org.apache.kafka.streams.test.OutputVerifier;
+import org.testng.annotations.AfterSuite;
+import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.Test;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNull;
+
+public class MegabusRefResolverTest {
+
+    // ref resolver stuff
+    private static final Topic refTopic = new Topic("refs", 1, (short) 1);
+    private static final Topic resolvedTopic = new Topic("resolvedRefs", 1, (short) 1);
+    private static final Topic retryTopic = new Topic("retry", 1, (short) 1);
+    private static final Topic missingRefsTopic = new Topic("missingRefs", 1, (short) 1);
+    private static final KafkaCluster kafkaCluster = mock(KafkaCluster.class);
+    private static final DataProvider dataProvider = new TestDataProvider();
+
+    static {
+        when(kafkaCluster.getBootstrapServers()).thenReturn("localhost:9092");
+    }
+
+    private static final MegabusRefResolver refResolver = new MegabusRefResolver(
+        dataProvider, refTopic, resolvedTopic, retryTopic, missingRefsTopic,
+        kafkaCluster, Clock.systemUTC(), HostAndPort.fromParts("localhost", 9092),
+        "refResolverGroup", new MetricRegistry()
+    );
+
+    // Kafka test driver stuff
+    private TopologyTestDriver testDriver;
+    private StringDeserializer stringDeserializer = new StringDeserializer();
+    private JsonPOJOSerde<List<MegabusRef>> jsonPOJOSerdeForMegabusList = new JsonPOJOSerde<>(new TypeReference<List<MegabusRef>>() {});
+    private JsonPOJOSerde<Map<String, Object>> jsonPOJOSerdeForMaps = new JsonPOJOSerde<>(new TypeReference<Map<String, Object>>() {});
+    private JsonPOJOSerde<MissingRefCollection> jsonPOJOSerdeForMissingRefCollection = new JsonPOJOSerde<>(new TypeReference<MissingRefCollection>() {});
+    private ConsumerRecordFactory<String, List<MegabusRef>> recordFactory = new ConsumerRecordFactory<>(new StringSerializer(), jsonPOJOSerdeForMegabusList.serializer());
+
+    @BeforeSuite
+    public void setUp() {
+        Properties config = new Properties();
+        config.setProperty(StreamsConfig.APPLICATION_ID_CONFIG, "refResolverGroup");
+        config.setProperty(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9092");
```

Comment:
would extract to `localhost:9092` to constant

Reply:
sure, although this value doesn't get used anywhere, it's just for interface compatibility.